### ISSUE TITLE
GTK 3.20 :: Lightdm :: Rewirte code

### ISSUE THREAD

**Issue by megax** [numixproject/numix-gtk-theme#391]
Example patch: https://github.com/The-Megax/Numix/commit/cccadf4736d698c7f0378c938d81211be34453fe

Test code:

```
/***********************
 ! LightDM GTK Greeter *
 ***********************/

@include exports("lightdm") {
    #panel_window {
        background-color: transparent;
        background-image: none;
        color: $white;
        font: bold;
        text-shadow: 0 1px alpha($black, .5);
        -gtk-icon-shadow: 0 1px alpha($black, .5);

        menubar, .menubar {
            &, > menuitem {
                border-style: none;
                background-color: transparent;
                background-image: none;
                color: $white;
                font: bold;
                text-shadow: 0 1px alpha($black, .5);
                -gtk-icon-shadow: 0 1px alpha($black, .5);

                *:hover { color: $white; }

                &:hover {
                    border-style: none;
                    background-color: alpha($white, .2);
                    background-image: none;
                    color: $white;
                }

                &:disabled { color: alpha($white, .7); }

                menu, .menu {
                    border-radius: 0;

                    menuitem {
                        font: normal;
                        text-shadow: none;
                    }
                }
            }
        }
    }

    #content_frame { padding-bottom: 14px; }

    #login_window, #shutdown_dialog, #restart_dialog {
        border-style: none;
        border-radius: $roundness;
        background-color: $lightdm_bg_color;
        color: $lightdm_fg_color;

        /* draw border using box-shadow */
        box-shadow: inset 1px 0 mix(shade($lightdm_bg_color, .7), $lightdm_fg_color, .21),
                    inset -1px 0 mix(shade($lightdm_bg_color, .7), $lightdm_fg_color, .21),
                    inset 0 1px mix(shade($lightdm_bg_color, .7), $lightdm_fg_color, .21),
                    inset 0 -1px mix(shade($lightdm_bg_color, .7), $lightdm_fg_color, .21);

        button {
            padding: 3px $spacing * 3;
            border-width: 1px;
            border-radius: $roundness;
            border-style: solid;
            border-color: shade($lightdm_bg_color, .8);
            background-color: shade($lightdm_bg_color, 1.08);
            background-image: none;
            color: $lightdm_fg_color;
            transition: all 150ms ease-out;

            &:focus, &:active:focus {
                border-color: shade($selected_bg_color, .8);
                background-color: shade($selected_bg_color, 1.08);
                background-image: none;
                color: $selected_fg_color;

                &:hover {
                    border-color: shade($selected_bg_color, .7);
                    background-color: $selected_bg_color;
                }
            }
        }
    }

    #login_window {
        menu, .menu { border-radius: 0; }

        combobox button {
            &, &:hover, &:active, &:active:hover,
            &:focus, &:hover:focus, &:active:focus, &:active:hover:focus {
                padding: 0;
                background: none;
                border-style: none;
                box-shadow: none;
            }
        }

        entry {
            padding: 3px $spacing;
            border-width: 1px;
            border-style: solid;
            border-color: shade($lightdm_bg_color, .8);
            border-radius: $roundness;
            background-color: shade($lightdm_bg_color, .9);
            background-image: none;
            color: $lightdm_fg_color;
            box-shadow: none;
            transition: all 150ms ease-out;

            &:focus, &:hover {
                border-color: shade($lightdm_bg_color, .7);

                box-shadow: inset 1px 0 alpha($dark_shadow, .1),
                            inset 0 1px alpha($dark_shadow, .12),
                            inset -1px 0 alpha($dark_shadow, .1),
                            inset 0 -1px alpha($dark_shadow, .05);
            }
        }
    }

    #user_combobox {
        color: $lightdm_fg_color;
        font: 18px;

        menu, .menu { font: normal; }

        arrow { color: mix($lightdm_fg_color, $lightdm_bg_color, .5); }
    }

    #user_image {
        padding: 3px;
        border-radius: $roundness;

        /* draw border using box-shadow */
        box-shadow: inset 1px 0 shade($lightdm_bg_color, .7),
                    inset -1px 0 shade($lightdm_bg_color, .7),
                    inset 0 1px shade($lightdm_bg_color, .7),
                    inset 0 -1px shade($lightdm_bg_color, .7);
    }

    #user_image_border {
        border-radius: $roundness;
        background-color: shade($lightdm_bg_color, .9);
        background-image: none;
        box-shadow: inset 1px 0 alpha($dark_shadow, .07),
                    inset 0 1px alpha($dark_shadow, .08),
                    inset -1px 0 alpha($dark_shadow, .07),
                    inset 0 -1px alpha($dark_shadow, .05);
    }

    #buttonbox_frame {
        padding-top: $spacing * 2;
        padding-bottom: 0;
        border-style: none;
        border-bottom-left-radius: $roundness;
        border-bottom-right-radius: $roundness;
        background-color: transparent;
        background-image: none;
        box-shadow: none;
    }

    /* shutdown button */
    #shutdown_button {
        border-color: shade($error_bg_color, .8);
        background-color: shade($error_bg_color, 1.08);
        background-image: none;
        color: $error_fg_color;

        &:hover, &:active, &:active:hover {
            border-color: shade($error_bg_color, .7);
            background-color: $error_bg_color;
        }
    }

    /* restart button */
    #restart_button {
        border-color: shade($warning_bg_color, .8);
        background-color: shade($warning_bg_color, 1.08);
        background-image: none;
        color: $warning_fg_color;

        &:hover, &:active, &:active:hover {
            border-color: shade($warning_bg_color, .7);
            background-color: $warning_bg_color;
        }
    }

     /* password warning */
    #greeter_infobar { font: bold; }
}
```

**Comment by lots0logs** [comment]
@megax I think the use of `#` symbol was dropped in 3.20, no?

**Comment by megax** [comment]
@lots0logs Currently used `#` symbol, not regression. (tested)

**Comment by lots0logs** [comment]
@megax Well there are no `#` symbols being used in selectors in Adwaita ~~so the styles in your commit are likely being ignored (well, those that use a `#` symbol in the selector at least)~~

~~https://github.com/GNOME/gtk/blob/master/gtk/theme/Adwaita/_common.scss~~

Nevermind..I see that they can still be used so long as the application assigns **CSS Names** to its widgets.

**Comment by khurshid-alam** [comment]
This has to wait. We need to test it against new lightdm which is still in proposed state in Yakkety.

**Comment by megax** [comment]
@khurshid-alam https://github.com/numixproject/numix-gtk-theme/pull/516

**Comment by lots0logs** [comment]
@megax LGTM except for one thing. I think the top panel should not be transparent as that is inconsistent with how panels are styled in the rest of the theme.

**Comment by megax** [comment]
@lots0logs However, I was amazed, why is It so. Modify :)

**Comment by lots0logs** [comment]
Closed by #516
